packages/core/helper-plugin/src/utils/getFetchClient.ts

```diff
@@ -1,16 +1,25 @@
+import { AxiosRequestConfig, AxiosResponse } from 'axios';
+
 import { instance } from './fetchClient';
 
-const addPrependingSlash = (url) =>
+const addPrependingSlash = (url: string) =>
   typeof url === 'string' && url.charAt(0) !== '/' ? `/${url}` : url;
 
 // This regular expression matches a string that starts with either "http://" or "https://" or any other protocol name in lower case letters, followed by "://" and ends with anything else
-const hasProtocol = (url) => new RegExp('^(?:[a-z+]+:)?//', 'i').test(url);
+const hasProtocol = (url: string) => new RegExp('^(?:[a-z+]+:)?//', 'i').test(url);
 
 // Check if the url has a prepending slash, if not add a slash
-const normalizeUrl = (url) => (hasProtocol(url) ? url : addPrependingSlash(url));
+const normalizeUrl = (url: string) => (hasProtocol(url) ? url : addPrependingSlash(url));
+
+type FetchClient = {
+  get: (url: string, config?: AxiosRequestConfig) => Promise<AxiosResponse>;
+  put: (url: string, data: any, config?: AxiosRequestConfig) => Promise<AxiosResponse>;
+  post: (url: string, data: any, config?: AxiosRequestConfig) => Promise<AxiosResponse>;
+  del: (url: string, config?: AxiosRequestConfig) => Promise<AxiosResponse>;
+};
 
-const getFetchClient = (defaultOptions = {}) => {
-  instance.defaults.baseURL = window.strapi.backendURL;
+const getFetchClient = (defaultOptions: AxiosRequestConfig = {}): FetchClient => {
+  instance.defaults.baseURL = (window as any).strapi.backendURL;
```

Comment:
I think we need to type `window.strapi`, there's a mock version of this type in `admin-test-utils`, imo we can start typing it an add more as we go, what do you think @markkaylor & @gu-stav?

Reply:
Because the entire monorepo isn't in TS yet and we're not able to share types across packages (for now) I would isolate it to the `helper-plugin` and as we move further out we can address how we're gonna share things :)